hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -74,6 +78,19 @@ class DefaultSource extends RelationProvider
     val tablePath = DataSourceUtils.getTablePath(fs, globPaths.toArray)
     log.info("Obtained hudi table path: " + tablePath)
 
+    val sparkEngineContext = new HoodieSparkEngineContext(sqlContext.sparkContext)
+    val fsBackedTableMetadata =
+      new FileSystemBackedTableMetadata(sparkEngineContext, new SerializableConfiguration(fs.getConf), tablePath, false)
+    val partitionPaths = fsBackedTableMetadata.getAllPartitionPaths
+    val onePartitionPath = if(!partitionPaths.isEmpty && !StringUtils.isEmpty(partitionPaths.get(0))) {
+        tablePath + "/" + partitionPaths.get(0)
+      } else {
+        tablePath
+      }
+    val dataPath = DataSourceUtils.getDataPath(tablePath, onePartitionPath)
+    log.info("Obtained hudi data path: " + dataPath)
+    parameters += "path" -> dataPath
```

Comment:
@teeyog Sorry it's not still clear to me. I supplied a globbed path `2015/*/*/*` and even that overrides `path -> tablePath/*/*/*/*` 

Won't this incur reading all partitions in the tablePath as opposed only 2015's? 

![image](https://user-images.githubusercontent.com/1179324/108234541-bde4fb00-70f9-11eb-8611-58579636b51b.png)

Reply:
The path specified by the user will be overwritten by the automatically inferred data directory, and your needs cannot be met

Reply:
I will try to see if I can automatically infer this but also meet your needs

Reply:
@vinothchandar Now it supports your needs. If the path specified by the user is a table path, it will be automatically inferred, otherwise it will not be inferred.